build/v1/types.go

```diff
@@ -1359,3 +1368,64 @@ type SecretSpec struct {
 }
+
+// BuildVolume describes a volume that is made available to build pods,
+// such that it can be mounted into buildah's runtime environment.
+// Only a subset of Kubernetes Volume sources are supported.
+type BuildVolume struct {
+	// name is a unique identifier for this BuildVolume.
+	// It must conform to the Kubernetes DNS label standard and be unique within the pod.
+	// It cannot collide with volumes that are always added by the build controller.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// source represents the location and type of the mounted volume.
+	Source BuildVolumeSource `json:"source" protobuf:"bytes,2,opt,name=source"`
+
+	// mounts represents the location and type of the mounted volume.
+	Mounts []BuildVolumeMount `json:"mounts" protobuf:"bytes,3,opt,name=mounts"`
+}
+
+// BuildVolumeSourceType represents a build volume source type
+type BuildVolumeSourceType string
+
+const (
+	// BuildVolumeSourceTypeSecret is the Secret build source volume type
+	BuildVolumeSourceTypeSecret BuildVolumeSourceType = "Secret"
+
+	// BuildVolumeSourceTypeConfigmap is the ConfigMap build source volume type
+	BuildVolumeSourceTypeConfigMap BuildVolumeSourceType = "ConfigMap"
+)
+
+// BuildVolumeSource represents the source of a volume to mount
+// Only one of its supported types may be specified at any given time.
+type BuildVolumeSource struct {
+
+	// type is the BuildVolumeSourceType for the volume source.
+	// Type must match the populated volume source, and if not specified will be inferred on admission.
+	// +k8s:conversion-gen=false
+	// +optional
+	Type BuildVolumeSourceType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=BuildVolumeSourceType"`
+
+	// secret represents a Secret that should populate this volume.
+	// More info: https://kubernetes.io/docs/concepts/storage/volumes#secret
+	// +optional
+	Secret *corev1.SecretVolumeSource `json:"secret,omitempty" protobuf:"bytes,2,opt,name=secret"`
+
+	// configMap represents a ConfigMap that should populate this volume
+	// +optional
+	ConfigMap *corev1.ConfigMapVolumeSource `json:"configMap,omitempty" protobuf:"bytes,3,opt,name=configMap"`
+}
+
+// BuildVolumeMount describes the mounting of a Volume within buildah's runtime environment.
+type BuildVolumeMount struct {
+	// readOnly denotes whether the volume should be mounted as readonly or not
+	// The volume is mounted read-only if true, read-write otherwise (false or unspecified).
+	// Defaults to false.
+	// +optional
+	ReadOnly bool `json:"readOnly,omitempty" protobuf:"varint,1,opt,name=readOnly"`
+
+	// destinationPath is the path within the build runtime environment at which the volume should be mounted.
+	// Must not contain ':' and can not collide with a destination path generated by the builder process.
+	DestinationPath string `json:"destinationPath" protobuf:"bytes,2,opt,name=destinationPath"`
```

Comment:
More of a nit - since `DestinationPath` is required, it should be listed first in the struct and protobuf.

```suggestion
	// destinationPath is the path within the build runtime environment at which the volume should be mounted.
	// Must not contain ':' and can not collide with a destination path generated by the builder process.
	DestinationPath string `json:"destinationPath" protobuf:"bytes,1,opt,name=destinationPath"`
	
	// readOnly denotes whether the volume should be mounted as readonly or not
	// The volume is mounted read-only if true, read-write otherwise (false or unspecified).
	// Defaults to false.
	// +optional
	ReadOnly bool `json:"readOnly,omitempty" protobuf:"varint,2,opt,name=readOnly"`
```

Reply:
updated in current push